build.gradle

```diff
@@ -234,28 +232,31 @@ task ktlint(type: JavaExec, group: "verification") {
 check.dependsOn ktlint
 check.dependsOn verifyPlugin
 
-def publishedProjects = subprojects.findAll { !it.name.contains("test")}
-
 task coverageReport(type: JacocoReport) {
     executionData fileTree(project.rootDir.absolutePath).include("**/build/jacoco/*.exec")
 
-    getAdditionalSourceDirs().from(publishedProjects.sourceSets.main.java.srcDirs)
-    getSourceDirectories().from(publishedProjects.sourceSets.main.java.srcDirs)
-    getClassDirectories().from(publishedProjects.sourceSets.main.output.classesDirs)
+    getAdditionalSourceDirs().from(subprojects.sourceSets.main.java.srcDirs)
+    getSourceDirectories().from(subprojects.sourceSets.main.java.srcDirs)
+    getClassDirectories().from(subprojects.sourceSets.main.output.classesDirs)
 
     reports {
         html.enabled true
         xml.enabled true
     }
 }
-publishedProjects.forEach {
+subprojects.forEach {
     coverageReport.mustRunAfter(it.tasks.withType(Test))
 }
 check.dependsOn coverageReport
 
 // Workaround for runIde being defined in multiple projects, if we request the root project runIde, "alias" it to
 // community edition
 if (gradle.startParameter.taskNames.contains("runIde")) {
+    // Only disable this if running from root project
+    if (gradle.startParameter.projectDir != project.rootProject.rootDir) {
+        return
+    }
```

Comment:
seems surprising to return from a top-level scope, but I assume this works somehow. (in Lua this would end the script, I guess gradle somehow skips the `if` scope and continues evaluating the rest of the script?)

Reply:
In Groovy, we are inside a Lambda so it just exits back to main scope

Reply:
It's odd looking, ill invert and wrap